src/app/components/showcase.tsx

```diff
@@ -1,6 +1,37 @@
+import { useEffect, useState } from 'react';
+import calculateScrollAlpha from '@/scripts/calculateScrollAlpha';
+
 export default function Showcase() {
+	const [alpha, setAlpha] = useState('0.00');
+	const [hidden, setHidden] = useState(true);
+
+	function handleScroll() {
+		let scrollAlpha: string = calculateScrollAlpha();
```

Comment:
I think this could probably be a const since it's not being reassigned? Also probably no need to cast it to a `: string` since the function itself is typed to return a string?

Reply:
Ahh so I was completely thinking about `let` and `const` in the wrong way in this context. I had originally created `scrollAlpha` as `let` since I thought it was getting continuously reassigned by `calculateScrollAlpha()`. But it's only getting its value assigned once per function call, so I see now how `const` works better in this case. Thanks for pointing that out!

So, maybe just beginner things with TypeScript, but in this case I was just trying to type everything I could. So, if I'm assigning a variable a set value, like `const text = "string of text"`, I don't need to cast it to a type? 

Or, should that be typed, and I only don't need to type `let scrollAlpha: string = calculateScrollAlpha();` since `calculateScrollAlpha()` already has a return type specified of string?

Reply:
For let vs const: yeah exactly, the `const` is scoped to the function itself, and it's never changing within that function call. You could even probably just get rid of your `hidden` state entirely also, and just check `alpha === '0.00'` where you need it. 

As for the typing, you're exactly right, the cast is unneeded since the function itself is already saying that it returns a string. Generally it's better to just let Typescript infer the types as long as it's able to do so correctly. So if your IDE can tell that the return value is a string, then you shouldn't need to explicitly declare that the variable is a string. But if typescript is saying the variable is `any` or `unknown`, then it's a sign you're missing a type declaration somewhere.

You can even enable a rule in your typescript config for `noImplicitAny` which will warn/error if there's anything with an ambiguous type. It can be a bit annoying at times because it's pretty strict, but it allows you to spot areas where the types are not well defined. 

This article goes into more depth about type inference with typescript: https://davidgomes.com/annotate-vs-type-inference/